Documentation update (mentions Melange and fixes a few things) (#765)
reasonml/reasonml.github.io: docs/what-and-why.md

title: What & Why
---

# Overview

> Reason is a user-friendly programming language built on the robust type system of [OCaml](http://ocaml.org), providing a comfortable transition for developers familiar with JavaScript or C-family languages.

Reviewer: ```suggestion
Reason is a user-friendly programming language built on [OCaml](http://ocaml.org)'s robust type system, making it a comfortable choice for developers familiar with JavaScript or C-family languages.
```

> Not only uses the type-system, but also integrates seamlessly with the reliable OCaml ecosystem, with access to all the tools from OCaml, such as: Language Server, the defacto build system [dune](https://dune.build), and the package manager [opam](https://opam.ocaml.org).

Reviewer: ```suggestion
Not only does Reason leverage the OCaml type system, but it also seamlessly integrates with its reliable ecosystem, providing access to all the tools of OCaml, including the Language Server, the de facto build system [Dune](https://dune.build), and the package manager [opam](https://opam.ocaml.org).
```

There's a clear distinction between the two environments, one runs JavaScript in the browser while the other runs assembly on the machine. But the language constructors and tooling is the same, so you can write code for the two environments.

**Compile to JavaScript**

[Melange](compiling-to-js-with-melange.md) is a backend for the OCaml compiler that emits JavaScript. Melange strives to provide the best integration with both the OCaml and JavaScript ecosystems. It is the recommended way to compile Reason to JavaScript, since it compiles 1 module to 1 file, has a very good interop with JavaScript and integrates well with modern frontend development tooling. Take a closer look at our page [Compiling to JavaSCript with Melange](compiling-to-js-with-melange.md).

**Compile to Native**

Native means using the OCaml compiler to produce native executables that can run directly on your machine. It's recommended to use [dune](https://dune.build) which is the defacto build system for native OCaml projects, but it's also possible to use [ocamlc (OCaml's compiler)](https://ocaml.org/docs/compiling-ocaml-projects) directly.

## Where do I start?

> The [getting started](getting-started.md) page will set you up

Reviewer: ```suggestion
The [getting started](getting-started.md) page will set you up.
```
